File: packages/config/src/features.ts
export const features = {
  tiza: {
    scannerGuided: true,
    itemAnalysis: true,
    studentProgress: true,
    chatSupport: true,
    billing: false,
    multiSchool: false,
  },
  relevo: {
    scannerGuided: false,
    bulkUpload: true,
    executiveKPIs: true,
    billing: true,
    multiSchool: true,
    whiteLabel: true,
  },
} as const;

export type Brand = 'tiza' | 'relevo';
export type FeatureFlags = typeof features.tiza | typeof features.relevo;

/** Todas las keys posibles de feature flags (merge de ambas marcas) */
export type AllFeatureKeys = keyof typeof features.tiza | keyof typeof features.relevo;

/** Mapa completo de feature flags con todas las keys posibles */
export type FeatureFlagsMap = { [K in AllFeatureKeys]: boolean };

/** Marca un FeatureFlags como FeatureFlagsMap para acceso completo a propiedades */
export function toFeatureFlagsMap(flags: FeatureFlags): FeatureFlagsMap {
  return flags as FeatureFlagsMap;
}

export function getFeatures(brand: Brand): FeatureFlags {
  return features[brand];
}

export function getFeaturesMap(brand: Brand): FeatureFlagsMap {
  return features[brand] as unknown as FeatureFlagsMap;
}

File: apps/api/services/features.py
"""Feature flags — brand defaults + tenant overrides."""

FEATURE_DEFAULTS = {
    "tiza": {
        "scannerGuided": True,
        "itemAnalysis": True,
        "studentProgress": True,
        "chatSupport": True,
        "billing": False,
        "multiSchool": False,
    },
    "relevo": {
        "scannerGuided": False,
        "bulkUpload": True,
        "executiveKPIs": True,
        "billing": True,
        "multiSchool": True,
        "whiteLabel": True,
    },
}


def get_features(brand: str, tenant_settings: dict | None = None) -> dict:
    """Resolve effective feature flags: brand defaults + tenant overrides.

    Args:
        brand: Brand name ("tiza" or "relevo").
        tenant_settings: Optional tenant settings dict (from Tenant.settings JSON).
                         If it contains a "features" key, those values override
                         the brand defaults.

    Returns:
        Merged feature flags dict.
    """
    flags = dict(FEATURE_DEFAULTS.get(brand, FEATURE_DEFAULTS["tiza"]))
    if tenant_settings and "features" in tenant_settings:
        flags.update(tenant_settings["features"])
    return flags
